driver-core/src/main/java/com/datastax/driver/core/policies/SpeculativeExecutionPolicy.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *      Copyright (C) 2012-2014 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Host;
+import com.datastax.driver.core.Statement;
+
+/**
+ * The policy that decides if the driver will send speculative queries to the next hosts when the current host takes too
```

Comment:
A minor comment: speculative execution does not take place "when" the host takes too long to respond. IMHO this should be something along the lines of : "...if the driver should send the same query to more than one host, to optimize response delays and defend itself against hosts that take too long to respond."
You could also point users to http://en.wikipedia.org/wiki/Speculative_execution.